main.tf

```diff
@@ -0,0 +1,377 @@
+locals {
+  # TODO: custom images
+  azure_common_image_aliases_json = jsondecode(file("${path.module}/azure_common_images.json"))
+
+  image = length(split(":", var.image)) == 4 ? {
+    publisher = split(":", var.image)[0]
+    offer     = split(":", var.image)[1]
+    sku       = split(":", var.image)[2]
+    version   = split(":", var.image)[3]
+  } : local.azure_common_image_aliases_json[index(local.azure_common_image_aliases_json[*].urnAlias, var.image)]
+
+  linux_offers = [
+    "0001-com-ubuntu-server-jammy",
+    "CentOS",
+    "debian-11",
+    "flatcar-container-linux-free",
+    "openSUSE-leap-15-4",
+    "RHEL",
+    "sles-15-sp3",
+  ]
+
+  windows_offers = [
+    "WindowsServer"
+  ]
```

Comment:
Build lists dynamically. Something like: if not `windows`, it is Linux.

Reply:
I could remember that some SQL server offerings exists.

Reply:
Implemented future-prove solution by detecting the operating system based on the image used, automatically with conditional `azapi_resource`.